dht_net.go

```diff
@@ -195,7 +203,11 @@ func (ms *messageSender) prep() error {
 		return err
 	}
 
-	ms.r = ggio.NewDelimitedReader(nstr, inet.MessageSizeMax)
+	r := ggio.NewDelimitedReader(nstr, inet.MessageSizeMax)
+	rch := make(chan chan requestResult, requestResultBuffer)
+	go messageReceiver(ms.dht.ctx, rch, r)
```

Comment:
We could save on long-running go routines by starting these only as needed and shutting them down when we have no more outstanding replies (i.e., by having some form of outstanding reply counter). If we don't keep these around, we could also afford to spin up a second goroutine to manage the outstanding requests.

Reply:
In general, I'm not a fan of pipelining but out-of-order replies would require a protocol change.

Reply:
Lazily spinning the goroutine is perhaps not so hard to implement, but shutting it down gets tricky -- not sure its worth the complexity. Let me think about it.